File: tmdb-core/src/main/java/dev/reuss/tmdb/domain/tv/series/model/TvSeriesEpisodeGroup.java
package dev.reuss.tmdb.domain.tv.series.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import dev.reuss.tmdb.common.TmdbModel;

/**
 * Episode group for a TMDB TV series.
 *
 * @param description  group description
 * @param episodeCount episode count
 * @param groupCount   group count
 * @param id           TMDB episode group id
 * @param name         group name
 * @param network      related network
 * @param type         group type
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public record TvSeriesEpisodeGroup(
        String description,

        @JsonProperty("episode_count")
        int episodeCount,

        @JsonProperty("group_count")
        int groupCount,

        String id,

        String name,

        TvSeriesNetwork network,

        int type
) implements TmdbModel {
}

File: tmdb-core/src/main/kotlin/dev/reuss/tmdb/domain/tv/series/model/TvSeriesEpisodeGroup.kt
package dev.reuss.tmdb.domain.tv.series.model

import com.fasterxml.jackson.annotation.JsonProperty
import dev.reuss.tmdb.common.TmdbModel
import dev.reuss.tmdb.domain.tv.episodegroup.model.TvEpisodeGroupType

/**
 * Episode group for a TMDB TV series.
 *
 * @property description group description
 * @property episodeCount episode count
 * @property groupCount group count
 * @property id TMDB episode group id
 * @property name group name
 * @property network related network
 * @property type group type
 */
@JvmRecord
data class TvSeriesEpisodeGroup(
    @all:JsonProperty("description")
    val description: String?,

    @all:JsonProperty("episode_count")
    val episodeCount: Int,

    @all:JsonProperty("group_count")
    val groupCount: Int,

    @all:JsonProperty("id")
    val id: String?,

    @all:JsonProperty("name")
    val name: String?,

    @all:JsonProperty("network")
    val network: TvSeriesNetwork?,

    @all:JsonProperty("type")
    val type: TvEpisodeGroupType
) : TmdbModel